client/engine/engine.go

```diff
@@ -721,3 +727,22 @@ func (e *Engine) recordMessageMetrics(message protocols.Message) {
 	e.metrics.RecordQueueLength(fmt.Sprintf("msg_payload_size,sender=%s,receiver=%s", e.store.GetAddress(), message.To), totalPayloadsSize)
 	e.metrics.RecordQueueLength(fmt.Sprintf("msg_size,sender=%s,receiver=%s", e.store.GetAddress(), message.To), len(raw))
 }
+
+func (e *Engine) checkError(err error) {
+	if err != nil {
+
+		e.logger.Err(err).Msgf("%s, error in run loop", e.store.GetAddress())
+
+		for _, nonFatalError := range nonFatalErrors {
+			if errors.Is(err, nonFatalError) {
+				return
+			}
+		}
+
+		<-time.After(1000 * time.Millisecond) // We wait for a bit so the previous log line has time to complete
+		// TODO do not panic if in production.
+		panic(err)
+		// TODO report errors back to the consuming application
```

Comment:
I think we could move this TODO to L738. More important to report errors that are (currently) being swallowed.

Reply:
I chose to rephrase this todo and place it above the relevant line (the panic) ac9aaf36

Reply:
Maybe I'm confused, but I think we don't necessarily want to say "instead of panic" now. I believe we've got a model now where we want _some_ errors to result in a panic. Probably all errors should be sent to the API consumer / user.